sky_sync/open_weather_api/comm.py

```diff
@@ -0,0 +1,132 @@
+from datetime import datetime
+import os
+
+import requests
+from requests import Response
+from requests.exceptions import ConnectionError, RequestException
+from rest_framework.exceptions import ParseError
+
+from .api_urls import (
+    build_current_weather_url,
+    build_direct_geocoding_url,
+    build_forecast_weather_url,
+    build_ip_url,
+    build_reverse_geocoding_url,
+)
+
+
+API_TOKEN = os.environ["OPEN_WEATHER_API_KEY"]
+
+
+def handle_request(url: str) -> Response:
+    """Handles GET requests with possible errors."""
+    try:
+        resp = requests.get(url=url)
+    except (RequestException, ConnectionError):
+        print("[bold red]An error occurred. Please check your network connection and try again.[/]")
+        exit(1)
+    return resp
+
+
+def get_current_weather_data(lat: float, lon: float) -> dict[str, str | float | dict]:
+    resp = handle_request(url=build_current_weather_url(lat=lat, lon=lon, units="metric", limit=5, appid=API_TOKEN))
+    data = resp.json()
+    if resp.status_code != 200:
+        raise ParseError(code=resp.status_code)
+    location_data = {
+        "name": data["name"],
+        "country": data["sys"]["country"],
+        "state": None,
+        "lat": data["coord"]["lat"],
+        "lon": data["coord"]["lon"],
+    }
+    rain: dict = data.get("rain", {})
+    snow: dict = data.get("snow", {})
+    current_data = {
+        "dt": datetime.fromtimestamp(data["dt"]),
```

Comment:
może ten proces parsowania do jakieś osobnej klasy przenieść? np. WeatherDataParser

Reply:
Podobnie, będę chciał to omówić.